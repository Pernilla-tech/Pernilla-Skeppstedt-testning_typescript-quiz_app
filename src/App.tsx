import React from "react";
import EndGame from "./components/endgame/EndGame";
import Game from "./components/Game";
import StartGame from "./components/StartGame/StartGame";

import config from "./confiq/QuizConfig";
import { useGameQuiz } from "./context/QuizGameContext";

function App() {
  const { screen, answers } = useGameQuiz();

  return (
    <div>
      {screen === "start" ? (
        <StartGame />
      ) : answers.length === config.totalQuestions ? (
        <EndGame />
      ) : (
        <Game />
      )}
    </div>
  );
}

export default App;
