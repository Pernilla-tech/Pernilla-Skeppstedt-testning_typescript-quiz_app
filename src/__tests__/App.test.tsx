import { render } from "@testing-library/react";
import App from "../App";

//creating a snapshot test to test if the rendered component is the same as the snapshot app
test("snapshot is correct", () => {
  const view = render(<App />);
  expect(view).toMatchSnapshot();
});
