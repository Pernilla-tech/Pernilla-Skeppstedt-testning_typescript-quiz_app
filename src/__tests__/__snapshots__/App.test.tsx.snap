// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot is correct 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="sc-gKPRtg cyMLoJ"
        >
          <div>
            <h1>
              Start quiz
            </h1>
            <input
              aria-label="Playername"
              data-testid="playername-input"
              placeholder="Enter your playername"
              type="text"
              value=""
            />
          </div>
          <p>
            Select difficulty
          </p>
          <select
            data-testid="select-difficulty"
          >
            <option
              value="easy"
            >
              easy
            </option>
            <option
              value="medium"
            >
              medium
            </option>
            <option
              value="hard"
            >
              hard
            </option>
            <option
              value="random"
            >
              random
            </option>
          </select>
          <p>
            Select region
          </p>
          <select
            data-testid="select-region"
          >
            <option
              value="SV"
            >
              SV
            </option>
            <option
              value="BG"
            >
              BG
            </option>
          </select>
          <button
            class="sc-gswNZR gkjvoA undefined"
            data-testid="start-quiz-button"
            disabled=""
          >
            Start quiz
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="sc-gKPRtg cyMLoJ"
      >
        <div>
          <h1>
            Start quiz
          </h1>
          <input
            aria-label="Playername"
            data-testid="playername-input"
            placeholder="Enter your playername"
            type="text"
            value=""
          />
        </div>
        <p>
          Select difficulty
        </p>
        <select
          data-testid="select-difficulty"
        >
          <option
            value="easy"
          >
            easy
          </option>
          <option
            value="medium"
          >
            medium
          </option>
          <option
            value="hard"
          >
            hard
          </option>
          <option
            value="random"
          >
            random
          </option>
        </select>
        <p>
          Select region
        </p>
        <select
          data-testid="select-region"
        >
          <option
            value="SV"
          >
            SV
          </option>
          <option
            value="BG"
          >
            BG
          </option>
        </select>
        <button
          class="sc-gswNZR gkjvoA undefined"
          data-testid="start-quiz-button"
          disabled=""
        >
          Start quiz
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
